packages/core/src/routes/interaction/index.test.ts

```diff
@@ -89,171 +102,191 @@ mockEsmDefault(
+      expect(verifyProfileSettings).toBeCalled();
+      expect(verifyIdentifierPayload).toBeCalled();
+      expect(storeInteractionResult).toBeCalled();
+      expect(response.status).toEqual(204);
     });
+  });
 
-    it('forgot password event should not call UserProfile validation', async () => {
-      const body = {
-        event: Event.ForgotPassword,
-        identifier: {
-          email: 'email@logto.io',
-          passcode: 'passcode',
-        },
-        profile: {
-          password: 'password',
-        },
-      };
-
-      const response = await sessionRequest.put(path).send(body);
-
-      expect(verifyIdentifier).toBeCalled();
-      expect(verifyProfile).toBeCalled();
-      expect(validateMandatoryUserProfile).not.toBeCalled();
-      expect(submitInteraction).toBeCalled();
-      expect(response.status).toEqual(200);
+  describe('DELETE /interaction', () => {
+    it('should call assignInteraction Result properly', async () => {
+      await sessionRequest.delete(`${interactionPrefix}`);
+      expect(assignInteractionResults).toBeCalled();
     });
   });
 
-  describe('PATCH /interaction', () => {
-    const path = interactionPrefix;
+  describe('PUT /interaction/event', () => {
+    const path = `${interactionPrefix}/event`;
 
-    afterEach(() => {
-      jest.clearAllMocks();
+    it('should call verifySignInModeSettings properly', async () => {
+      getInteractionStorage.mockResolvedValueOnce({
+        event: Event.SignIn,
+      });
+      const body = {
+        event: Event.Register,
+      };
+
+      const response = await sessionRequest.put(path).send(body);
+      expect(getInteractionStorage).toBeCalled();
+      expect(verifySignInModeSettings).toBeCalled();
+      expect(storeInteractionResult).toBeCalled();
+      expect(response.status).toEqual(204);
     });
 
-    it('sign-in event with register event interaction session in record should call methods properly', async () => {
-      getInteractionStorage.mockResolvedValueOnce({ event: Event.Register });
+    it('should reject if switch sign-in event to forgot-password directly', async () => {
+      getInteractionStorage.mockResolvedValueOnce({
+        event: Event.SignIn,
+      });
 
       const body = {
-        event: Event.SignIn,
+        event: Event.ForgotPassword,
       };
 
-      const response = await sessionRequest.patch(path).send(body);
-      expect(verifyIdentifier).toBeCalled();
-      expect(verifyProfile).toBeCalled();
-      expect(validateMandatoryUserProfile).toBeCalled();
-      expect(submitInteraction).toBeCalled();
-      expect(response.status).toEqual(200);
-      expect(response.body).toEqual({ redirectUri: 'logto.io' });
+      const response = await sessionRequest.put(`${interactionPrefix}/event`).send(body);
+      expect(verifySignInModeSettings).toBeCalled();
+      expect(storeInteractionResult).not.toBeCalled();
+      expect(response.status).toEqual(404);
```

Comment:
Should this be a 404 error? I don't think so... maybe 422?

Reply:
the idea is like indicating the forgot password interaction, not exit.